### PR TITLE
Convert circle.class.ts to Typescript

### PR DESCRIPTION
**Pull request by avra-m3**

### CODE REVIEW

**Comment by asturur** [comment]
Hi @avra-m3  you are having the issue we are all having.
87 tests in the browser fail.
We are looking in what we did wrong.

**Comment by avra-m3** [comment]
No worries, I figured it wasn't related to this pr - is the code style one also meant to be failing?

**Comment by asturur** [comment]
@avra-m3 there are 2 things to take in account.
1. types: we want to discuss internally who to actually do them ( for example yes, cacheproperties are an array of string, but we would like them to be an enum of those the possible properties )
2. things are WIP in this moment and breaks easily

https://github.com/fabricjs/fabric.js/pull/8118
This is how i converted Rect, and from contributors we are looking forwad prs of this kind.
Transform the old code in more modern es6+ code, deconstructs whatever you can find.

One of the issues i had was that i naturally used lfat arrows for one liners, but that was wrong and i didn't expect because usually i have layers of babel that fix that.

You can ask all the support you want if you want to help.

**Comment by asturur** [comment]
Oh LINT is supposed to fail, but not on the main file you are working on.

**Comment by asturur** [comment]
Another tip, i know this code well and i was making plain mistakes in the previous PR, and then i started this new own, and i did baby steps.

I suggest you start fresh from master, and give a look at the commits i did, do the White space commit at the end when everything pass.

**Comment by avra-m3** [comment]
Sorry, took initiative on the types thinking it would save work on a subsequent pass, happy to roll them all back, or is it mostly just the cache properties one?

**Comment by avra-m3** [comment]
I just checked your pr and either I missed some of your changes on the first pass or you made some updates - I'll fixup this pr tomorrow. Apologies.

**Comment by asturur** [comment]
> Sorry, took initiative on the types thinking it would save work on a subsequent pass, happy to roll them all back, or is it mostly just the cache properties one?

In my message i mispeleed how with who.
There is not to execuse for initiative, is fine, is just that in this moment would be just more code changes. We want to have all the library converted to es6 and import/exports before start typing or refactoring or adding types.

**Comment by asturur** [comment]
> I just checked your pr and either I missed some of your changes on the first pass or you made some updates - I'll fixup this pr tomorrow. Apologies.

The PR you saw was probably the one i closed in favor of this.

**Comment by avra-m3** [comment]
> > Sorry, took initiative on the types thinking it would save work on a subsequent pass, happy to roll them all back, or is it mostly just the cache properties one?
> 
> In my message i mispeleed how with who.
> There is not to execuse for initiative, is fine, is just that in this moment would be just more code changes. We want to have all the library converted to es6 and import/exports before start typing or refactoring or adding new features.

Yep easy totally understand, I'll leave it to es6 changes, I didn't get that part of the memo. I assume leave in the ts nocheck comment at the top in that case?

Thanks for the quick replies sorry for the misunderstandings.

**Comment by ShaMan123** [comment]
@avra-m3 if you need help don't hesitate to ping one of us

File: src/shapes/circle.class.ts [hunk 1]
```diff
@@ -1,201 +1,200 @@
...
+/**
+ * Returns {@link fabric.Circle} instance from an SVG element
+ * @static
+ * @memberOf fabric.Circle
+ * @param {SVGElement} element Element to parse
+ * @param {Function} [callback] Options callback invoked after parsing is finished
+ * @throws {Error} If value of `r` attribute is missing or invalid
+ */
+Circle.fromElement = function(element: SVGElement, callback: (result: typeof Circle) => void) {
+  const parsedAttributes = fabric.parseAttributes(element, fabric.Circle.ATTRIBUTE_NAMES);
```
Comment (ShaMan123) [inline src/shapes/circle.class.ts:169]:
```suggestion
  const parsedAttributes = fabric.parseAttributes(element, Circle.ATTRIBUTE_NAMES);
```

lookup all fabric.Circle and replace them

**Review by ShaMan123 — CHANGES_REQUESTED** [review]

**Comment by avra-m3** [comment]
Declining and will raise a pr from new branch.

**Pull request closed without merging**